### ISSUE TITLE
Linux support?

### ISSUE THREAD

**Issue by Impeta** [DiscordMessenger/dm#8]
I wouldn't mind to build it myself even, I'm just a little lost on instruction right now. Could you rephrase it for us Linux users kindly? Thanks!

**Comment by iProgramMC** [comment]
Discord Messenger does not support Linux. I'm not sure what you want. Do you want to compile it with MinGW on Linux?

**Comment by iProgramMC** [comment]
Well, @er2off has managed to compile it with MinGW, although I think he switched the project over to CMake which I don't really feel like doing.  Essentially it consists of replacing Windows includes to all lowercase (`<ShellAPI.h>` to `<shellapi.h>` for example), editing the makefile to not try to run exes, and it should work

**Comment by iProgramMC** [comment]
Let me know if you can compile it following similar steps as MinGW on Windows, now that I updated all windows includes to be lowercase.

**Comment by er2off** [comment]
> now that I updated all windows includes to be lowercase.

@iProgramMC you forgot to update submodules.
Attaching all of my local [patches.zip](https://github.com/DiscordMessenger/dm/files/15278770/patches.zip) because WTF GITHUB WHY I CAN'T UPLOAD PATCH

**Comment by iProgramMC** [comment]
I may have neglected to update my submodules indeed. I'll take care of that asap.

**Comment by iProgramMC** [comment]
@er2off I've added your gesture changes, and lowercased all Windows standard includes.  Could you maybe try compiling with MinGW on Linux again?

**Comment by iProgramMC** [comment]
Well yeah, the makefile is specifically made for MinGW 6.3.0 on Windows. I should make it so you can override FIND and MKDIR (so `MKDIR ?=` instead of `MKDIR =`). This will allow you to use `MKDIR=mkdir FIND=find make`

**Comment by username00112233** [comment]
I _think_ DM hardcodes the interface using the WINAPI (windows.h)

So for linux support you would have to rewrite it in QT or something

**Comment by averyterrel** [comment]
I don't think that's a goal of this project, since any supported QT versions don't support Windows 2000, and certainly not the recently added Windows 95 target. You're probably better off running this under WINE.

**Comment by 0944-tw** [comment]
It's not really possible. The other things successfully compiled. but windows part isn't.
It relies on WINAPI so much, No QT, No GTK, Only WinAPI

**Comment by iProgramMC** [comment]
Of course. You need MinGW to compile it, even if you compile it from Linux for Windows.

**Comment by averyterrel** [comment]
> or may it natively linked to libwine ?

Just run it on wine. Official Linux support is not planned any time soon.

**Comment by sir-octopus** [comment]
At the risk of necro posting, works great on macOS Sequoia 15 and Wine 11.
